### ISSUE TITLE
macOS linker_messages `__eh_frame section too large`; false positive?

### ISSUE THREAD

**Issue by carols10cents** [rust-lang/rust#159105]
## What happened

After upgrading to 1.97 on macOS, I'm now getting this `linker_messages` warning:

```
warning: linker stderr: ld: __eh_frame section too large (max 16MB) to encode dwarf unwind offsets in compact unwind table, performance of exception handling might be affected
  |
  = note: `#[warn(linker_messages)]` on by default

warning: `crates_io` (bin "background-worker") generated 1 warning
```

I have no idea what this means or if I'm supposed to do anything about this. I read the release blog post that said some of these are false positives that should be filtered out, so I'm thinking this might be one of those?

## How to reproduce

I get this consistently when building the `background-worker` bin of crates.io.

```
$ git clone https://github.com/rust-lang/crates.io.git
$ cd crates.io
$ cargo build --bin background-worker
```

## System information

I'm on an Apple M1 Max using Sequoia 15.7.7.

```
$ rustc --version --verbose
rustc 1.97.0 (2d8144b78 2026-07-07)
binary: rustc
commit-hash: 2d8144b7880597b6e6d3dfd63a9a9efae3f533d3
commit-date: 2026-07-07
host: aarch64-apple-darwin
release: 1.97.0
LLVM version: 22.1.6
```
